File: boilerplateFE/src/constants/permissions.ts
/**
 * Frontend mirror of Starter.Shared.Constants.Permissions.
 *
 * Pattern: {Module}.{Action}
 *
 * Standard CRUD actions:
 *   View   -> list / read-many
 *   Show   -> read single / detail
 *   Create -> create resource
 *   Update -> update resource
 *   Delete -> delete resource
 *
 * Keep this file in sync with the backend Permissions.cs.
 */
export const PERMISSIONS = {
  Users: {
    View: 'Users.View',
    Show: 'Users.Show',
    Create: 'Users.Create',
    Update: 'Users.Update',
    Delete: 'Users.Delete',
    ManageRoles: 'Users.ManageRoles',
  },
  Roles: {
    View: 'Roles.View',
    Show: 'Roles.Show',
    Create: 'Roles.Create',
    Update: 'Roles.Update',
    Delete: 'Roles.Delete',
    ManagePermissions: 'Roles.ManagePermissions',
  },
  Tenants: {
    View: 'Tenants.View',
    Show: 'Tenants.Show',
    Create: 'Tenants.Create',
    Update: 'Tenants.Update',
    Delete: 'Tenants.Delete',
  },
  Files: {
    View: 'Files.View',
    Upload: 'Files.Upload',
    Delete: 'Files.Delete',
    Manage: 'Files.Manage',
    ShareOwn: 'Files.ShareOwn',
  },
  ApiKeys: {
    View: 'ApiKeys.View',
    Create: 'ApiKeys.Create',
    Update: 'ApiKeys.Update',
    Delete: 'ApiKeys.Delete',
    ViewPlatform: 'ApiKeys.ViewPlatform',
    CreatePlatform: 'ApiKeys.CreatePlatform',
    UpdatePlatform: 'ApiKeys.UpdatePlatform',
    DeletePlatform: 'ApiKeys.DeletePlatform',
    EmergencyRevoke: 'ApiKeys.EmergencyRevoke',
  },
  FeatureFlags: {
    View: 'FeatureFlags.View',
    Create: 'FeatureFlags.Create',
    Update: 'FeatureFlags.Update',
    Delete: 'FeatureFlags.Delete',
    ManageTenantOverrides: 'FeatureFlags.ManageTenantOverrides',
    OptOut: 'FeatureFlags.OptOut',
  },
  Billing: {
    View: 'Billing.View',
    Manage: 'Billing.Manage',
    ViewPlans: 'Billing.ViewPlans',
    ManagePlans: 'Billing.ManagePlans',
    ManageTenantSubscriptions: 'Billing.ManageTenantSubscriptions',
  },
  Products: {
    View: 'Products.View',
    Create: 'Products.Create',
    Update: 'Products.Update',
  },
  Comments: {
    View: 'Comments.View',
    Create: 'Comments.Create',
    Edit: 'Comments.Edit',
    Delete: 'Comments.Delete',
    Manage: 'Comments.Manage',
  },
  Activity: {
    View: 'Activity.View',
  },
  System: {
    ViewAuditLogs: 'System.ViewAuditLogs',
    ManageSettings: 'System.ManageSettings',
    ViewDashboard: 'System.ViewDashboard',
    ExportData: 'System.ExportData',
    ForceExport: 'System.ForceExport',
    ImportData: 'System.ImportData',
  },
  Communication: {
    View: 'Communication.View',
    ManageChannels: 'Communication.ManageChannels',
    ManageIntegrations: 'Communication.ManageIntegrations',
    ManageTemplates: 'Communication.ManageTemplates',
    ManageTriggerRules: 'Communication.ManageTriggerRules',
    ViewDeliveryLog: 'Communication.ViewDeliveryLog',
    Resend: 'Communication.Resend',
    ManageQuotas: 'Communication.ManageQuotas',
  },
  Webhooks: {
    View: 'Webhooks.View',
    Create: 'Webhooks.Create',
    Update: 'Webhooks.Update',
    Delete: 'Webhooks.Delete',
    ViewPlatform: 'Webhooks.ViewPlatform',
  },
  Ai: {
    Chat: 'Ai.Chat',
    ViewConversations: 'Ai.ViewConversations',
    DeleteConversation: 'Ai.DeleteConversation',
    ManageAssistants: 'Ai.ManageAssistants',
    ManageDocuments: 'Ai.ManageDocuments',
    ManageTools: 'Ai.ManageTools',
    ManageTriggers: 'Ai.ManageTriggers',
    ViewUsage: 'Ai.ViewUsage',
    RunAgentTasks: 'Ai.RunAgentTasks',
    ManageSettings: 'Ai.ManageSettings',
    SearchKnowledgeBase: 'Ai.SearchKnowledgeBase',
    RunEval: 'Ai.RunEval',
  },
  Workflows: {
    View: 'Workflows.View',
    ManageDefinitions: 'Workflows.ManageDefinitions',
    Start: 'Workflows.Start',
    ActOnTask: 'Workflows.ActOnTask',
    Cancel: 'Workflows.Cancel',
    ViewAllTasks: 'Workflows.ViewAllTasks',
  },
} as const;

type PermissionMap = typeof PERMISSIONS;
export type Permission = {
  [M in keyof PermissionMap]: PermissionMap[M][keyof PermissionMap[M]];
}[keyof PermissionMap];

File: boilerplateMobile/lib/core/permissions/permissions.dart
/// Permission constants mirroring the backend's
/// `Starter.Shared/Constants/Permissions.cs`.
///
/// Keep in sync manually — when the BE adds/removes permissions,
/// update this file. Module-specific permissions (e.g. Billing) are
/// declared in their respective module's constants file, not here.
abstract final class Permissions {
  // --- Users ---
  static const usersView = 'Users.View';
  static const usersShow = 'Users.Show';
  static const usersCreate = 'Users.Create';
  static const usersUpdate = 'Users.Update';
  static const usersDelete = 'Users.Delete';
  static const usersManageRoles = 'Users.ManageRoles';

  // --- Roles ---
  static const rolesView = 'Roles.View';
  static const rolesShow = 'Roles.Show';
  static const rolesCreate = 'Roles.Create';
  static const rolesUpdate = 'Roles.Update';
  static const rolesDelete = 'Roles.Delete';
  static const rolesManagePermissions = 'Roles.ManagePermissions';

  // --- System ---
  static const systemManageSettings = 'System.ManageSettings';
  static const systemViewDashboard = 'System.ViewDashboard';
  static const systemViewAuditLogs = 'System.ViewAuditLogs';
  static const systemExportData = 'System.ExportData';
  static const systemForceExport = 'System.ForceExport';

  // --- Tenants ---
  static const tenantsView = 'Tenants.View';
  static const tenantsShow = 'Tenants.Show';
  static const tenantsCreate = 'Tenants.Create';
  static const tenantsUpdate = 'Tenants.Update';
  static const tenantsDelete = 'Tenants.Delete';

  // --- Files ---
  static const filesView = 'Files.View';
  static const filesUpload = 'Files.Upload';
  static const filesDelete = 'Files.Delete';
  static const filesManage = 'Files.Manage';

  // --- Feature Flags ---
  static const featureFlagsView = 'FeatureFlags.View';
  static const featureFlagsCreate = 'FeatureFlags.Create';
  static const featureFlagsUpdate = 'FeatureFlags.Update';
  static const featureFlagsDelete = 'FeatureFlags.Delete';
  static const featureFlagsManageTenantOverrides =
      'FeatureFlags.ManageTenantOverrides';
  static const featureFlagsOptOut = 'FeatureFlags.OptOut';

  // --- API Keys ---
  static const apiKeysView = 'ApiKeys.View';
  static const apiKeysCreate = 'ApiKeys.Create';
  static const apiKeysUpdate = 'ApiKeys.Update';
  static const apiKeysDelete = 'ApiKeys.Delete';
  static const apiKeysViewPlatform = 'ApiKeys.ViewPlatform';
  static const apiKeysCreatePlatform = 'ApiKeys.CreatePlatform';
  static const apiKeysUpdatePlatform = 'ApiKeys.UpdatePlatform';
  static const apiKeysDeletePlatform = 'ApiKeys.DeletePlatform';
  static const apiKeysEmergencyRevoke = 'ApiKeys.EmergencyRevoke';
}
